compiler/z/codegen/FPTreeEvaluator.cpp

```diff
@@ -1204,86 +1204,119 @@ OMR::Z::TreeEvaluator::fbits2iEvaluator(TR::Node * node, TR::CodeGenerator * cg)
-   deps->addPostCondition(sourceReg, TR::RealRegister::AssignAny);
-   generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, cleansedNumber, deps);
-   generateRXInstruction(cg, TR::InstOpCode::STE, node, sourceReg, tempMR);
-   if((TR::Compiler->target.cpu.getS390SupportsFPE()) && (!disabled))
-      {
-      TR::Register *targetReg = cg->allocate64bitRegister();
-      generateRRInstruction(cg, TR::InstOpCode::LGDR, node, targetReg, sourceReg);
-      generateRSInstruction(cg, TR::InstOpCode::SRLG, node, targetReg, targetReg, 32);
-      node->setRegister(targetReg);
-      cg->decReferenceCount(node->getFirstChild());
-      if (node->getNumChildren() == 2)
-         cg->decReferenceCount(node->getSecondChild());
-      return targetReg;
-      }
+   //Disable FPE Version For Now Due To Performance Reasons
+   bool disableFPInstructions = true;
+
+   // if child is a load with refCount == 1, then just use a regular load
+   // to load as an integer 
+   if (node->getFirstChild()->getRegister() == NULL &&
+      node->getFirstChild()->getOpCode().isLoadVar() &&
+      (node->getFirstChild()->getReferenceCount() == 1))
+      {
+      targetReg = cg->allocateRegister();
+      tempmemref = generateS390MemoryReference(node->getFirstChild(), cg);
+      targetReg = genericLoadHelper<32, 32, MemReg>(node, cg, tempmemref, NULL, false, true);
+      }
+    else
+       {
+       sourceReg = cg->evaluate(node->getFirstChild());
+       if((TR::Compiler->target.cpu.getS390SupportsFPE()) && (!disableFPInstructions))
+          {
+          TR::Register *targetReg = cg->allocate64bitRegister();
+          // Load into GR from FPR
+          generateRRInstruction(cg, TR::InstOpCode::LGDR, node, targetReg, sourceReg);
+
+          // TODO: Why is this needed? Is the data in the FPR instruction above loaded into the 
+          // top half of the targetReg? PoPs doesn't say much about this instruction. 
+          generateRSInstruction(cg, TR::InstOpCode::SRLG, node, targetReg, targetReg, 32);
+          }
+          else
+          {
+          // regular GPR approach
+          TR::MemoryReference * tempMR1 = generateS390MemoryReference(node, f2iSR, cg);
+          generateRXInstruction(cg, TR::InstOpCode::STE, node, sourceReg, tempMR1);
+
+          // TODO: Is there a difference between MR1 and MR2?
+          TR::MemoryReference * tempMR2 = generateS390MemoryReference(node, f2iSR, cg);
+          targetReg = genericLoadHelper<32, 32, MemReg>(node, cg, tempMR2, NULL, false, true);
+          // TODO: I think we can remove this as it's handled automatically for us. 
+          tempMR2->stopUsingMemRefRegister(cg); 
+          }
+       }
+    
+    //TODO: for an fbits2i node, this should always be set. However in my unit test these instructions
+    // never get generated. I believe the same problem occurs on X. We need to investigate why. 
+    // This should always be set for a java program.
+    if (node->normalizeNanValues())
```

Comment:
@fjeremic The x86 evaluator has a similar guard. As can be seen by this comment: 
https://github.com/eclipse/openj9/issues/890#issuecomment-367805534, we never generated the normalization instructions and we should have (this is for a java program). 

I checked inside openj9, and we set this normalization parameter to true when we inline the function call here: https://github.com/eclipse/openj9/blob/b2b62046eb4c221ad1773457bae20da74f08ec09/runtime/compiler/env/VMJ9.cpp#L7813-L7818

So it should always be true for a java program.